test/position-managers/PositionManager.notifier.t.sol

```diff
@@ -0,0 +1,283 @@
+            );
+        }
+
+        bytes memory calls = plan.finalizeModifyLiquidity(config.poolKey);
+        lpm.modifyLiquidities(calls, _deadline);
+
+        assertEq(sub.notifySubscribeCount(), 1);
+        assertEq(sub.notifyModifyLiquidityCount(), 10);
+    }
+
+    function test_notifyTransfer_withTransferFrom_succeeds() public {
+        uint256 tokenId = lpm.nextTokenId();
+        mint(config, 100e18, alice, ZERO_BYTES);
+
+        // approve this contract to operate on alices liq
+        vm.startPrank(alice);
+        lpm.approve(address(this), tokenId);
+        vm.stopPrank();
+
+        lpm.subscribe(tokenId, config, address(sub));
+
+        lpm.transferFrom(alice, bob, tokenId);
+
+        assertEq(sub.notifyTransferCount(), 1);
+    }
+
+    function test_notifyTransfer_withSafeTransferFrom_succeeds() public {
+        uint256 tokenId = lpm.nextTokenId();
+        mint(config, 100e18, alice, ZERO_BYTES);
+
+        // approve this contract to operate on alices liq
+        vm.startPrank(alice);
+        lpm.approve(address(this), tokenId);
+        vm.stopPrank();
+
+        lpm.subscribe(tokenId, config, address(sub));
+
+        lpm.safeTransferFrom(alice, bob, tokenId);
+
+        assertEq(sub.notifyTransferCount(), 1);
+    }
+
+    function test_notifyTransfer_withSafeTransferFromData_succeeds() public {
+        uint256 tokenId = lpm.nextTokenId();
+        mint(config, 100e18, alice, ZERO_BYTES);
+
+        // approve this contract to operate on alices liq
+        vm.startPrank(alice);
+        lpm.approve(address(this), tokenId);
+        vm.stopPrank();
+
+        lpm.subscribe(tokenId, config, address(sub));
+
+        lpm.safeTransferFrom(alice, bob, tokenId, "");
+
+        assertEq(sub.notifyTransferCount(), 1);
+    }
+
+    function test_unsubscribe_succeeds() public {
+        uint256 tokenId = lpm.nextTokenId();
+        mint(config, 100e18, alice, ZERO_BYTES);
+
+        // approve this contract to operate on alices liq
+        vm.startPrank(alice);
+        lpm.approve(address(this), tokenId);
+        vm.stopPrank();
+
+        lpm.subscribe(tokenId, config, address(sub));
+
+        lpm.unsubscribe(tokenId, config);
+
+        assertEq(sub.notifyUnsubscribeCount(), 1);
+    }
+
+    function test_unsubscribe_isSuccessfulWithBadSubscriber() public {
+        uint256 tokenId = lpm.nextTokenId();
+        mint(config, 100e18, alice, ZERO_BYTES);
+
+        // approve this contract to operate on alices liq
+        vm.startPrank(alice);
+        lpm.approve(address(this), tokenId);
+        vm.stopPrank();
+
+        lpm.subscribe(tokenId, config, address(badSubscriber));
+
+        MockReturnDataSubscriber(badSubscriber).setReturnDataSize(0x600000);
+        lpm.unsubscribe(tokenId, config);
+
+        // the subscriber contract call failed bc it used too much gas
+        assertEq(MockReturnDataSubscriber(badSubscriber).notifyUnsubscribeCount(), 0);
```

Comment:
check that the tokenId no longer has a subscriber to show it was successful

Reply:
`assertEq(address(lpm.subscriber(tokenId)), address(sub));`

Reply:
and check that the 1/0 in config is set correctly too

Reply:
throughout this file i'd love to see checks that the `subscriber[]` and the bit are setting and unsetting when they should

Reply:
good call, added